### ISSUE TITLE
Wrong scope in condition without braces with destructor

### ISSUE THREAD

**Issue by tupkalov** [babel/babel#5023]
### Input Code

```js
var context = {
  firstMethod () {
    
  	var innerMethod = () => {
  		if(condition)
  			({ destructor } = this.secondMethod());
  	};
  
  	innerMethod()
  
  },
  secondMethod () {
    
  }
}
```

### Output Code

```js
"use strict";

var context = {
  firstMethod: function firstMethod() {

    var innerMethod = function innerMethod() {

      if (condition) {
        ;

        var _secondMethod = this.secondMethod();

        destructor = _secondMethod.destructor;
      }
    };

    innerMethod();
  },
  secondMethod: function secondMethod() {}
};
```

### Babel Configuration (.babelrc, package.json, cli command)

```js
{
  "presets": ["es2015"]
}
```

### Expected Behavior
In output line with `this.secondMethod()` this must be _this or _this2 and etc.

### Current Behavior
In output line with `this.secondMethod()` this === window

### Your Environment
Babel "Try it out" page

**Comment by hzoo** [comment]
more minimal repro is

```
var innerMethod = () => {
	if (true)
		({ destructor } = this);
};
 ```

**Comment by hzoo** [comment]
Might also be the same issue as https://github.com/babel/babel/issues/4363 @danez?

**Comment by danez** [comment]
maybe, i should look at this again.

**Comment by udnisap** [comment]
I was troubleshooting this for a bit and found out that the destructuring assignment variable resolution is having the issue. in the following example, I called the method before the inline if block and that got resolved with the correct context. 

**Input code**
```js
var context = {
  firstMethod () {
    
  	var innerMethod = () => {
  	  this.secondMethod()
  		if(condition)
  		  	({ destructor } = this.secondMethod());
  		
  	};
  
  	innerMethod()
  
  },
  secondMethod () {
    
  }
}
```
**Output code**
```js
"use strict";

var context = {
  firstMethod: function firstMethod() {
    var _this = this;

    var innerMethod = function innerMethod() {
      _this.secondMethod();
      if (condition) {
        ;

        var _secondMethod = this.secondMethod();

        destructor = _secondMethod.destructor;
      }
    };

    innerMethod();
  },
  secondMethod: function secondMethod() {}
};
```

**Comment by buunguyen** [comment]
Proposed fix in https://github.com/babel/babel/pull/5743

**Comment by danez** [comment]
This was resolved with logans refactoring around scope-hoisting in #5677
This is fixed in the current 7.0 alpha version and will be part of 7.0 final.
